src/couch/src/couch_query_servers.erl

```diff
@@ -247,6 +248,30 @@ sum_arrays([X|Xs], [Y|Ys]) when is_number(X), is_number(Y) ->
 sum_arrays(Else, _) ->
     throw_sum_error(Else).
 
+check_sum_overflow(InSize, OutSize, Sum) ->
+    Overflowed = OutSize > 4906 andalso OutSize * 2 > InSize,
+    case config:get("query_server_config", "reduce_limit", "true") of
+        "true" when Overflowed ->
+            Msg = log_sum_overflow(InSize, OutSize),
+            {[
+                {<<"error">>, <<"builtin_reduce_error">>},
+                {<<"reason">>, Msg}
+            ]};
+        "log" when OutSize > 4096 andalso OutSize > 2 * InSize ->
```

Comment:
Use `Overflowed` variable here as well, it also the updated correct logic for overflow.

To double check:

https://github.com/apache/couchdb/blob/bf72d619d11d787c4cd5bf343d7dfeb4b16ef01a/share/server/views.js#L41

Reply:
Ah, good catch. I refactored that out intending to have it both places.